WordPress/Classes/Services/ReaderSiteService.m

```diff
@@ -86,7 +87,7 @@ - (void)followSiteWithID:(NSUInteger)siteID success:(void(^)())success failure:(
             if (success) {
                 success();
             }
-            [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed];
+            [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed withProperties:@{ WPAppAnalyticsKeyBlogID:[NSNumber numberWithUnsignedInteger:siteID] }];
```

Comment:
I'd replace `[NSNumber numberWithUnsignedInteger:siteID]` with the newer syntactic sugar `@(siteID)`